### PR TITLE
[20] Fix NPE with two objects with same id but different eClass

### PR DESCRIPTION
**Pull request by florianbarbin**
Bug: https://github.com/eclipse-emf-compare/emf-compare/issues/20

### CODE REVIEW

File: plugins/org.eclipse.emf.compare/src/org/eclipse/emf/compare/match/DefaultMatchEngine.java [hunk 1]
```diff
@@ -119,9 +128,120 @@ public Comparison match(IComparisonScope scope, Monitor monitor) {
...
+
+	/**
+	 * Verifies if the compared elements share the same eClass. If it is not the case, we log an error in the
+	 * diagnostic.
+	 * 
+	 * @param match
+	 *            The match that is to be checked.
+	 * @param monitor
+	 *            The monitor to report progress or to check for cancellation.
+	 * @return true if the compared elements share the same eClass, false otherwise.
```
Comment (Kellindil) [inline plugins/org.eclipse.emf.compare/src/org/eclipse/emf/compare/match/DefaultMatchEngine.java:156]: no @return here

File: plugins/org.eclipse.emf.compare/src/org/eclipse/emf/compare/match/DefaultMatchEngine.java [hunk 2]
```diff
@@ -547,4 +667,8 @@ public static IEObjectMatcher createDefaultEObjectMatcher(UseIdentifiers useIDs,
 
 		return matcher;
 	}
+
+	private static final class EClassDiagnostic {
+
+	}
```
Comment (Kellindil) [inline plugins/org.eclipse.emf.compare/src/org/eclipse/emf/compare/match/DefaultMatchEngine.java:673]: To be deleted

**Review by Kellindil — CHANGES_REQUESTED** [review]

File: plugins/org.eclipse.emf.compare/src/org/eclipse/emf/compare/match/DefaultMatchEngine.java (hunk 1 above)
Comment (florianbarbin) [reply plugins/org.eclipse.emf.compare/src/org/eclipse/emf/compare/match/DefaultMatchEngine.java:156]: done

File: plugins/org.eclipse.emf.compare/src/org/eclipse/emf/compare/match/DefaultMatchEngine.java (hunk 2 above)
Comment (florianbarbin) [reply plugins/org.eclipse.emf.compare/src/org/eclipse/emf/compare/match/DefaultMatchEngine.java:673]: done

**Pull request merged**